Tests/AK/TestStringUtils.cpp

```diff
@@ -407,6 +407,14 @@ TEST_CASE(find)
     EXPECT_EQ(AK::StringUtils::find(test_string, "78"sv).has_value(), false);
 }
 
+TEST_CASE(replace_all_overlapping)
+{
+    // Replace only should take into account non-overlapping instances of the needle,
+    // since it is looking to replace them.
+    DeprecatedString test_string = "\\\\";
+    EXPECT_EQ(AK::StringUtils::replace(test_string, "\\"sv, "-"sv, ReplaceMode::All), "--"sv);
```

Comment:
Can we add some of the longer runs from the Jakt issue just to be sure :D

Reply:
OK, I'm grabbing Jakt now so it spits the stuff before the replacement should take care, because I'm not sure of what the final unescaped strings were meant to be 😅

Reply:
I think when I wrote that issue I was trying to embed quoted strings in the c++ block that I was generating from jakt? Or something equally cursed.